### ISSUE TITLE
ElasticSearch Datasource doesn't provide data when creating alerts

### ISSUE THREAD

**Issue by gabor** [grafana/grafana#61848]
### Discussed in https://github.com/grafana/grafana/discussions/59692

<div type='discussions-op-text'>

<sup>Originally posted by **tleuschner** December  2, 2022</sup>
I configured a ElasticSearch datasource which seems to be working properly in the dashboards panel and the explore tab. However when trying to create an alert with the ElasticSearch datasource, there seems to be a problem.
First I get the following error, when initially selecting the datasource
```
Failed to evaluate queries and expressions: [plugin.downstreamError] failed to query data: type assertion to string failed
```
After that, when changing query, the error seems to disappear but no data is returned.

Do you have any idea on how to fix this issue?

Setup:
grafana Version: 9.2.5
Elastic Search Datasource
- SigV4 Auth using the AWS SDK Default
- ElasticSearch version 7.10+

Kind regards
</div>

**Comment by gabor** [comment]
hi @tleuschner , i moved the problem you reported in https://github.com/grafana/grafana/discussions/59692 into a github-isuse, should be easier to handle it.

1. could you please try with newest grafana version?
2. could you try out something in the dashboard?:
    - create a dashboard panel with the elastic query
    - this should work correctly
    - now add an [Expression] (button in the lower-left corner), set it's value to `$A` (this means: take the value of the query `A`. `A` is by default the name of your first query)
    - make the query run (click refresh-button or something else).
    - does this work? or are you getting the same problem as in alerting?
    - ( see screenshot in https://github.com/grafana/grafana/issues/60614#issuecomment-1387313782 )

**Comment by gabor** [comment]
issue may be related to https://github.com/grafana/grafana/issues/60614

**Comment by kwenzh** [comment]
the same promblem:
![image](https://user-images.githubusercontent.com/38367518/221738893-04a7b620-0df4-427a-babd-b7389489112f.png)

**Comment by kwenzh** [comment]
in grafana 9.0.7, use es datasource alert
![image](https://user-images.githubusercontent.com/38367518/221741966-2e39fb60-c97a-42dd-a47c-172228b8862e.png)

RawData is no data
![image](https://user-images.githubusercontent.com/38367518/221742038-932ccb03-ed5b-4314-90c9-dec4920ab34e.png)

**Comment by gabor** [comment]
hi @kwenzh , thanks for the extra info, sorry for the long delay.

the fact that "raw data" mode does not work is a known limitation. 
we did a lot of work in grafana lately to make the "alerting" mode for elastic more feature-complete, so raw-data shoul work here when Grafana 10 happens.  (i don't know if you will be able to configure an useful alert based on the returned data, but you will get data returned).

to "restart" our discussion, is it only in "raw data mode" where you get "no data", or does it happen with "date histogram" too? (if it does, could you post a screenshot showing the query-editor configuration)?

(also, grafana 9.0.7 is quite an old version, i do recommend trying this with a newer version)

**Comment by kwenzh** [comment]
> hi @kwenzh , thanks for the extra info, sorry for the long delay.
> 
> the fact that "raw data" mode does not work is a known limitation. we did a lot of work in grafana lately to make the "alerting" mode for elastic more feature-complete, so raw-data shoul work here when Grafana 10 happens. (i don't know if you will be able to configure an useful alert based on the returned data, but you will get data returned).
> 
> to "restart" our discussion, is it only in "raw data mode" where you get "no data", or does it happen with "date histogram" too? (if it does, could you post a screenshot showing the query-editor configuration)?
> 
> (also, grafana 9.0.7 is quite an old version, i do recommend trying this with a newer version)

Was the first fixed in version 10.x?

**Comment by ivanahuckova** [comment]
> Was the first fixed in version 10.x?

I think 9.5 should have a fix as well, but 10.0 has it for sure.
